Engine/ac/string.cpp

```diff
@@ -257,6 +258,55 @@ const char* String_UpperCase(const char *thisString) {
     return CreateNewScriptString(std::move(buf));
 }
 
+void * String_Split(const char *thisString, const char *separator)
+{
+    if ((thisString == nullptr) || (thisString[0] == 0))
+        return nullptr;
+
+    auto const &header = ScriptString::GetHeader((void*)thisString);
+    std::vector<const char*> items{};
+
+    if ((separator == nullptr) || (separator[0] == 0)) {
+        items.push_back(thisString);
+        DynObjectRef arr = DynamicArrayHelpers::CreateStringArray(items);
+        return arr.Obj;
+    }
+
+    size_t seplen = strlen(separator);
+    if (seplen > header.Length) {
+        items.push_back(thisString);
+        DynObjectRef arr = DynamicArrayHelpers::CreateStringArray(items);
+        return arr.Obj;
+    }
+
+    // we will use this list to hold copies of substrings, they will be cleared when this functions exits
+    std::vector<ScriptString::Buffer> buflist = {};
+    const char *ptr = thisString;
+    const char *end = thisString + header.Length;
+    size_t len = 0;
+    while (ptr < end)
+    {
+        const char *found_cstr = strstr(ptr, separator);
+        if (!found_cstr) {
+            found_cstr = end;
+        }
+
+        len = found_cstr - ptr;
+        auto buf = ScriptString::CreateBuffer(len, 0);
+        std::copy(ptr, found_cstr, buf.Get());
+        buf.Get()[len] = 0;
```

Comment:
I think this could be simplified by using a single `snprintf(buf.Get(), "%s", ptr, len + 1);`

This also makes me wonder why I haven't added an assignment operator to ScriptString::Buffer, which would just copy an input string over...